vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextFieldVariant.java

```diff
@@ -15,26 +15,28 @@
  */
 package com.vaadin.flow.component.textfield;
 
+import com.vaadin.flow.component.ThemeVariant;
+
 /**
- * Set of theme variants applicable for {@code vaadin-text-field} component.
+ * Set of theme variants applicable for the {@link TextField} component, as well
+ * as other components based on it.
  */
-public enum TextFieldVariant {
-    LUMO_SMALL("small"), LUMO_ALIGN_CENTER("align-center"), LUMO_ALIGN_RIGHT(
-            "align-right"), LUMO_HELPER_ABOVE_FIELD(
-                    "helper-above-field"), MATERIAL_ALWAYS_FLOAT_LABEL(
-                            "always-float-label");
+public enum TextFieldVariant implements ThemeVariant {
+    //@formatter:off
+    LUMO_SMALL("small"),
+    LUMO_ALIGN_CENTER("align-center"),
+    LUMO_ALIGN_RIGHT("align-right"),
+    LUMO_HELPER_ABOVE_FIELD("helper-above-field"),
+    MATERIAL_ALWAYS_FLOAT_LABEL("always-float-label");
+    //@formatter:on
 
     private final String variant;
 
     TextFieldVariant(String variant) {
         this.variant = variant;
     }
 
-    /**
-     * Gets the variant name.
-     *
-     * @return variant name
-     */
+    @Override
     public String getVariantName() {
         return variant;
     }
```

Comment:
I guess I'm not very familiar with how incapsulation works in Java, but I wonder if this part is required? It seems to be just a repetition of `ThemeVariant`.